build.sbt

```diff
@@ -267,52 +116,25 @@ lazy val commonSettings = Seq(
 )
 
 lazy val commonJvmSettings = Seq(
-  testOptions in Test += Tests.Argument(TestFrameworks.ScalaTest, "-oDF")
+  Test / testOptions += Tests.Argument(TestFrameworks.ScalaTest, "-oDF")
 )
 
 lazy val commonJsSettings = Seq(
-  scalaJSStage in Global := FastOptStage,
+  Global / scalaJSStage := FastOptStage,
   parallelExecution := false,
   jsEnv := new org.scalajs.jsenv.nodejs.NodeJSEnv(),
   coverageEnabled := false
 )
 
 lazy val commonNativeSettings = Seq(
-  crossScalaVersions := crossScalaVersions.value.filter(_.startsWith("2.")),
+  crossScalaVersions := (ThisBuild / crossScalaVersions).value.filter(_.startsWith("2.")),
+  // Remove when native is published for the default previous versions
+  tlVersionIntroduced := List("2.12", "2.13").map(_ -> "0.4.1").toMap,
   coverageEnabled := false
 )
 
-def previousArtifact(version: String, proj: String) = {
-  def mod(x: Int, y: Int, z: Int): ModuleID =
-    "org.typelevel" %% s"paiges-$proj" % s"$x.$y.$z"
-
-  // the "-dbuild..." part is for Scala community build friendliness
-  val regex = "([0-9]+)\\.([0-9]+)\\.([0-9]+)(-SNAPSHOT|-dbuild[a-z0-9]*|\\+.*)?".r
-  version match {
-    case regex(smajor, sminor, spatch, suffix) =>
-      val (major, minor, patch) = (smajor.toInt, sminor.toInt, spatch.toInt)
-
-      // unless we're in a 0.x release, we need to ensure that our
-      // minor version is compatible with previous minor versions.
-      //
-      // for example, 4.1.1 should be compatible with 4.1.0 and also
-      // with 4.0.0.
-      //
-      // ideally we'd want to ensure that 4.1.1 was compatible with
-      // the latest 4.0.x release (e.g. 4.0.13) but that would require
-      // parsing our full release history; currently the algorithm is
-      // only based on the current version.
-      val minors = if (major > 0) (0 until minor).toSet else Set.empty
-      val patches = (0 until patch).toSet
-      val current = if (suffix != null && suffix.startsWith("+")) Set(mod(major, minor, patch)) else Set.empty[ModuleID]
-      minors.map(mod(major, _, 0)) | patches.map(mod(major, minor, _)) | current
-    case _ =>
-      throw new RuntimeException(s"Could not parse Paiges version: $version")
-  }
-}
-
 lazy val noPublish = commonSettings ++ Seq(
-  skip in publish := true,
+  publish / skip := true,
   mimaPreviousArtifacts := Set.empty,
   coverageEnabled := false
 )
```

Comment:
We can remove these settings, anywhere you'd want them use `NoPublishPlugin` instead.